test/Solnet.Programs.Test/InstructionDecoderTest.cs

```diff
@@ -0,0 +1,631 @@
+            Assert.AreEqual(0, decodedInstructions[2].InnerInstructions.Count);
+            Assert.IsTrue(decodedInstructions[2].Values.TryGetValue("Offset", out object offset));
+            Assert.IsTrue(decodedInstructions[2].Values.TryGetValue("Data", out object data));
+            Assert.AreEqual(35UL, (ulong)offset);
+            CollectionAssert.AreEqual(new byte[]{ 1, 0, 15, 25, 77,}, (byte[])data);
+        }
+
+        [TestMethod]
+        public void DecodeDurableNonceMessageTest()
+        {
+            Message msg = Message.Deserialize(DurableNonceMessage);
+            List<DecodedInstruction> decodedInstructions = InstructionDecoder.DecodeInstructions(msg);
+
+            Assert.AreEqual(2, decodedInstructions.Count);
+            Assert.AreEqual("Advance Nonce Account", decodedInstructions[0].InstructionName);
+            Assert.AreEqual("System Program", decodedInstructions[0].ProgramName);
+            Assert.AreEqual("11111111111111111111111111111111", decodedInstructions[0].PublicKey);
+            Assert.AreEqual(0, decodedInstructions[0].InnerInstructions.Count);
+            Assert.IsTrue(decodedInstructions[0].Values.TryGetValue("Nonce Account", out object nonceAccount));
+            Assert.IsTrue(decodedInstructions[0].Values.TryGetValue("Authority", out object authority));
+            Assert.AreEqual("G5EWCBwDM5GzVNwrG9LbgpTdQBD9PEAaey82ttuJJ7Qo", (PublicKey)nonceAccount);
+            Assert.AreEqual("5omQJtDUHA3gMFdHEQg1zZSvcBUVzey5WaKWYRmqF1Vj", (PublicKey)authority);
+        }
+        
+        [TestMethod]
+        public void DecodeCreateAccountWithSeedTest()
+        {
+            Message msg = Message.Deserialize(CreateWithSeedTransferCheckedMessage);
+            List<DecodedInstruction> decodedInstructions = InstructionDecoder.DecodeInstructions(msg);
+
+            Assert.AreEqual(4, decodedInstructions.Count);
+            Assert.AreEqual("Create Account With Seed", decodedInstructions[0].InstructionName);
+            Assert.AreEqual("System Program", decodedInstructions[0].ProgramName);
+            Assert.AreEqual("11111111111111111111111111111111", decodedInstructions[0].PublicKey);
+            Assert.AreEqual(0, decodedInstructions[0].InnerInstructions.Count);
+            Assert.IsTrue(decodedInstructions[0].Values.TryGetValue("From Account", out object fromAccount));
+            Assert.IsTrue(decodedInstructions[0].Values.TryGetValue("To Account", out object toAccount));
+            Assert.IsTrue(decodedInstructions[0].Values.TryGetValue("Base Account", out object baseAccount));
+            Assert.IsTrue(decodedInstructions[0].Values.TryGetValue("Seed", out object seed));
+            Assert.IsTrue(decodedInstructions[0].Values.TryGetValue("Space", out object space));
+            Assert.IsTrue(decodedInstructions[0].Values.TryGetValue("Amount", out object amount));
+            Assert.IsTrue(decodedInstructions[0].Values.TryGetValue("Owner", out object ownerAccount));

```

Comment:
Comment for future reference:
In cases like this, were you have multiple steps before the last assert, you don't need to assert every step of the way if its easily traceable the origin of the problem.
For instance, in this case, you could've missed the `IsTrue` asserts for the `TryGetValue`, because for each of the `AreEqual` bellow to be true, its guaranteed that the `TryGetValue` had to succeed.

Another example would be
```
Assert.NotNull(foo);
Assert.IsTrue(foo.Bar);
```

If `foo` was ever null, you'd find out right in the next assert with a NPE, so the `Assert.NotNull` becomes redundant, and kinda pollutes the unit test. 

Not an issue tho, just saved keystrokes in the future :)